src/service/grpc.cpp

```diff
@@ -54,33 +54,33 @@ grpc::ServerUnaryReactor *Grpc::CreateAccessPolicy(
 
 	for (const auto &principal : request->principals()) {
 		try {
+			datastore::Identities identities;
+
 			switch (principal.type()) {
 			case gk::v1::PrincipalType::collection:
 				policy.addCollectionPrincipal(principal.id());
+				identities = datastore::ListIdentitiesInCollection(principal.id());
```

Comment:
🤔 Do we need to retrieve full identities here? If we just get the IDs it'll be a PK lookup in `collections_identities`, which will be more efficient than having to do a join and then also return all the Identity data when we aren't really using them.